homeassistant/components/melnor/__init__.py

```diff
@@ -0,0 +1,56 @@
+"""The melnor integration."""
+
+
+from melnor_bluetooth.device import Device
+
+from homeassistant.components import bluetooth
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DEFAULT_CONNECTION_TIMEOUT, DOMAIN
+from .models import MelnorDataUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up melnor from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {}).setdefault(entry.entry_id, {})
+
+    ble_device = bluetooth.async_ble_device_from_address(hass, entry.data[CONF_ADDRESS])
+
+    # Create the device and connect immediately so we can pull down
+    # required attributes before building out our entities
+    device = Device(entry.data[CONF_ADDRESS], ble_device)
+    await device.connect(timeout=DEFAULT_CONNECTION_TIMEOUT)
+
+    if not device.is_connected:
+        raise ConfigEntryNotReady(f"Failed to connect to: {device.mac}")
+
+    coordinator = MelnorDataUpdateCoordinator(hass, device)
+
+    hass.data[DOMAIN][entry.entry_id]["coordinator"] = coordinator
```

Comment:
If coordinator is the only thing you are storing in the entry, there is no need to create a `dict` to store it. You can store it directly

Also if you need a `dict`, please use a `dataclass` instead

Reply:
I've simplified this (in a commit that I have yet to push) to store the coordinator directly.